alf/algorithms/mcts_algorithm.py

```diff
@@ -453,9 +586,13 @@ def _build_tree2(self, trees, to_plays):
         trees.value_sum[roots] += branch_factor * v
         self._update_best_child(trees, roots)
 
+        max_depth = torch.zeros(trees.B.shape[0], dtype=torch.int64)
```

Comment:
There is already `self._max_depth` and this naming seems to have a conflict. Would something like `longest_path` (and `avg_path_length`) or `deepest` work?